runtime/common/src/crowdloan.rs

```diff
@@ -482,23 +484,27 @@ decl_module! {
 				Error::<T>::InRetirementPeriod
 			);
 
-			let account = Self::fund_account_id(index);
-
-			// Avoid using transfer to ensure we don't pay any fees.
-			let transfer = WithdrawReasons::TRANSFER;
-			let imbalance = T::Currency::withdraw(&account, fund.deposit, transfer, AllowDeath)?;
-			let _ = T::Currency::resolve_into_existing(&fund.owner, imbalance);
+			// Try killing the crowdloan child trie
+			match Self::crowdloan_kill(index) {
+				child::KillOutcome::AllRemoved => {
+					let account = Self::fund_account_id(index);
+					T::Currency::transfer(&account, &fund.owner, fund.deposit, AllowDeath)?;
 
-			let imbalance = T::Currency::withdraw(&account, fund.raised, transfer, AllowDeath)?;
-			T::OrphanedFunds::on_unbalanced(imbalance);
+					// Remove all other balance from the account into orphaned funds.
+					let (imbalance, _) = T::Currency::slash(&account, BalanceOf::<T>::max_value());
+					T::OrphanedFunds::on_unbalanced(imbalance);
 
-			Self::crowdfund_kill(index);
-			<Funds<T>>::remove(index);
+					<Funds<T>>::remove(index);
 
-			Self::deposit_event(RawEvent::Dissolved(index));
+					Self::deposit_event(RawEvent::Dissolved(index));
+				},
+				child::KillOutcome::SomeRemaining => {
```

Comment:
There must be an explanation for this that I haven't reached yet, but I can't help but ask: where/when is the rest of this being deleted, if we happen to encounter it? I wasn expecting something similar to contracts pallet where we gradually wipe on_initialize.

Reply:
I think the sender is expected to send this dispatchable until is is completely resolved in order to get the deposit back. We need to make sure that the deposit is high enough so that the sender is incentivized to do so.

Reply:
sensible, thanks. 

> We need to make sure that the deposit is high enough so that the sender is incentivized to do so.

👍

Reply:
Anyone can call this extrinsic, so as long as someone wants to maintain the network, it will be called. Even if it costs someone some tokens to do it, you can imagine Parity or Web3 calling this to clean up storage on the blockchain.